New Pattern Submission - TranslateDocument s3-lambda-translate-cdk-python (#2049)
aws-samples/serverless-patterns: s3-lambda-translate-cdk-python/README.md

> # S3 -> Lambda -> Translate

Reviewer: ```suggestion
# Translate file with AWS Lambda, Amazon S3, and Amazon Translate
```

## Setup

### Set up environment and gather packages

```
cd s3-lambda-translate-cdk-python
```

Install the required dependencies (aws-cdk-lib and constructs) into your Python environment 
```
pip install -r requirements.txt
```

### Gather and deploy resources with the CDK

First synthesize, which executes the application, defines which resources will be created, and translates this into a CloudFormation template
```
cdk synth
```
All AWS CDK v2 deployments use dedicated AWS resources to hold data during deployment. Therefore, your AWS account and Region must be bootstrapped to create these resources before you can deploy. If you haven't already bootstrapped execute the below command
```
cdk bootstrap
```
and deploy with
```
cdk deploy
```

The deployment will create two S3 buckets and a Lambda function.

## How it works
The S3 bucket acts as a placeholder to upload the document, required for performing language translation. In the demonstration, we use the file `AmazonSimpleStorageService.html` inside the `assets` folder. 

> When the file is uploaded to input S3 bucket, the Lambda function is triggered. 

Reviewer: ```suggestion
Uploading a file to the S3 bucket invokes the Lambda function.
```

> The Lambda function invokes the Translate's `TranslateDocument` API  and the response document with the nomenclature `target_language`-`source_file_name` is uploaded to the output S3 bucket. 

Reviewer: ```suggestion
The Lambda function invokes Translate's `TranslateDocument` API  and uploads the  response document with the naming pattern `target_language`-`source_file_name`. 
```

The target language for translation is set to French by default and users are requested to change it as per their use-case.
At the time of creating this pattern, `TranslateDocument` API supports three formats of document:
   - `text/html` - The input data consists of HTML content. Amazon Translate translates only the text in the HTML element.
   - `text/plain` -  The input data consists of unformatted text. Amazon Translate translates every character in the content.
   - `application/vnd.openxmlformats-officedocument.wordprocessingml.document` - The input data consists of a Word document (.docx).

In this demonstration, we've chosen the `text/html` document format.

## Testing
Upon successful deployment of the stack, the Output section would provide the names of the S3 buckets from the variables `S3InputBucket` and `S3OutputBucket` in the CDK environment. 
Alternatively, these values can be found from the Output section of the `CloudFormation` stack.

> Upload the sample file `assets\AmazonSimpleStorageService.html` into the input S3 bucket. The upload action triggers the Lambda function and the document is analyzed. 

Reviewer: ```suggestion
Upload the sample file `assets\AmazonSimpleStorageService.html` to the input S3 bucket. The upload action invokes the Lambda function and the document is analyzed. 
```

> The translated document is stored in the output S3 bucket. In this demonstration, we have chosen to use a html document to convert it to `French` language. 

Reviewer: ```suggestion
The translated document is stored in the output S3 bucket. In this demonstration, we have chosen to convert an html document to `French. 
```
